pr-queue/index.js

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+/* Check if the trigger was a PR */
+function is_pull_request() {
+  return "GITHUB_BASE_REF" in process.env;
+}
+
+/* Get the target branch to check for PRs against */
+function target_branch() {
+  if (is_pull_request()) {
+    return process.env.GITHUB_BASE_REF;
+  } else {
+    return process.env.GITHUB_REF;
+  }
+}
+
+/* Get the name of the PR branch that triggered the PR */
+function pr_branch() {
+  if (is_pull_request()) {
+    return process.env.GITHUB_HEAD_REF;
+  }
+}
+
+/* Split the repository name into owner and repository */
+function repo_split() {
+  const repo = process.env.GITHUB_REPOSITORY;
+  const split_index = repo.indexOf('/');
+  return [repo.substr(0, split_index), repo.substr(split_index + 1)];
+}
+
+/* Get the owner of the repository */
+function repo_owner() {
+  return repo_split(0);
+}
+
+/* Get the name of the repository */
+function repo_name() {
+  return repo_split(1);
+}
+
+/* Determine if a PR is ahead of its target */
+async function is_ahead(octokit, pr) {
+  const comparison = await octokit.repos.compareCommits({
+    owner: repo_owner(),
+    repo: repo_name(),
+    base: pr.base.label,
+    head: pr.head.label,
+  });
+
+  return comparison.status == "ahead";
+}
+
+/* Determine if a PR has been approved */
+async function is_approved(octokit, pr) {
+  const reviews = await octokit.pulls.listReviews({
+    owner: repo_owner(),
+    repo: repo_name(),
+    pull_number: pr.number,
+  });
+
+  const is_approved = review => review.state == "APPROVED";
+  return reviews.filter(is_approved).length > 0;
```

Comment:
We'd usually want 2 approvals. (1 is possible, but should be an exception, e.g. if the PR is from someone who already has commit rights).

Reply:
This is more a prompt to get any reviewers at all, the actual number of reviewers should be checked by the assignee who then goes to merge.

Reply:
Wouldn't we want to prefer a PR that already has all the necessary reviewers? I guess one approval should still beat no approvals, though. Happy to see first if it makes any difference in practice if you want.